File: mobile/src/main/java/treehou/se/habit/ui/settings/SettingsContract.java
package treehou.se.habit.ui.settings;

import treehou.se.habit.R;
import treehou.se.habit.mvp.BasePresenter;
import treehou.se.habit.mvp.BaseView;

public interface SettingsContract {

    interface View extends BaseView<Presenter> {
        void showWidgetSettings();
        void showGeneralSettings();
        void showLicense();
        void showTranslatePage();
    }

    interface Presenter extends BasePresenter {
        void openWidgetSettings();
        void openGeneralSettings();
        void openLicense();
        void openTranslatePage();
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/SettingsContract.kt
package treehou.se.habit.ui.settings

import treehou.se.habit.R
import treehou.se.habit.mvp.BasePresenter
import treehou.se.habit.mvp.BaseView

interface SettingsContract {

    interface View : BaseView<Presenter> {
        fun showGeneralSettings()
        fun showLicense()
        fun showTranslatePage()
    }

    interface Presenter : BasePresenter {
        fun openGeneralSettings()
        fun openLicense()
        fun openTranslatePage()
    }
}
